src/Promise.fs

```diff
@@ -138,6 +141,13 @@ module Promise =
         member x.Using<'T, 'R when 'T :> IDisposable>(resource: 'T, binder: 'T->JS.Promise<'R>): JS.Promise<'R> =
             x.TryFinally(binder(resource), fun () -> resource.Dispose())
 
+        [<CustomOperation("and!", IsLikeZip=true)>]
+        member x.Merge(a: JS.Promise<'T1>, b: JS.Promise<'T2>, [<ProjectionParameter>] resultSelector : 'T1 -> 'T2 -> 'R): JS.Promise<'R> =
+            bind (fun (result: obj[]) ->
+                match result with
+                | [|a';b'|] -> lift (resultSelector (unbox<'T1> a') (unbox<'T2> b'))
+                | _ -> !!JS.Promise.reject(sprintf "Expected two results, but received %i." result.Length)) (Parallel [|map box a; map box b|])

```

Comment:
See the previous commit 8000b3242342746decff060d8b6ba6318b82079b for an implementation that avoids explicit boxing.